js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useTimeRangeFilter.tsx

```diff
@@ -12,33 +13,43 @@ import {useUpdatingRef} from '../../hooks/useUpdatingRef';
 
 import {FilterObject, FilterTag, FilterTagHighlightedText} from './useFilter';
 
+dayjs.extend(utc);
+dayjs.extend(timezone);
+
 import 'react-dates/initialize';
 import 'react-dates/lib/css/_datepicker.css';
 
 export type TimeRangeState = [number | null, number | null];
 
 export function calculateTimeRanges(timezone: string) {
+  const targetTimezone = timezone === 'Automatic' ? browserTimezone() : timezone;
+  const nowTimestamp = Date.now();
+  const now = Math.floor(nowTimestamp);
+  const startOfDay = dayjs(nowTimestamp).tz(targetTimezone).startOf('day');
   const obj = {
     TODAY: {
       label: 'Today',
-      range: [moment().tz(timezone).startOf('day').toDate().valueOf(), null] as TimeRangeState,
+      range: [startOfDay.valueOf(), now] as TimeRangeState,
     },
     YESTERDAY: {
       label: 'Yesterday',
       range: [
-        moment().tz(timezone).subtract(1, 'day').startOf('day').toDate().valueOf(),
-        moment().tz(timezone).subtract(1, 'day').endOf('day').toDate().valueOf(),
+        dayjs(nowTimestamp).tz(targetTimezone).subtract(1, 'day').startOf('day').valueOf(),
+        startOfDay.valueOf(),
       ] as TimeRangeState,
     },
     LAST_7_DAYS: {
       label: 'Within last 7 days',
-      range: [moment().tz(timezone).subtract(7, 'days').toDate().valueOf(), null] as TimeRangeState,
+      range: [
+        dayjs(nowTimestamp).tz(targetTimezone).subtract(1, 'week').startOf('day').valueOf(),
```

Comment:
The "startOf(day)" bit here is new, but it seems broadly reasonable to include. I wonder if we want this to show exactly 168 hours though?

Reply:
ah yeah, let me update this.

Btw, re: edit: I wonder if the underlying issue was that we were passing Automatic through to moment as the timezone? that seems to be a functional change here.

I thought the same, but when I tested that locally it worked fine (I use 'Automatic' locally)